theano/tensor/nnet/__init__.py

```diff
@@ -151,3 +152,86 @@ def conv2d(input, filters, input_shape=None, filter_shape=None,
                            border_mode, subsample, filter_flip,
                            filter_dilation)
+
+
+def conv2d_transpose(input, filters, output_shape, filter_shape=None,
+                     border_mode='valid', input_dilation=(1, 1),
+                     filter_flip=True, filter_dilation=(1, 1)):
+    """
+    This function will build the symbolic graph for applying a transposed
+    convolution over a mini-batch of a stack of 2D inputs with a set of 2D
+    filters.
+
+
+    Parameters
+    ----------
+    input: symbolic 4D tensor
+        Mini-batch of feature map stacks, of shape
+        (batch size, input channels, input rows, input columns).
+        See the optional parameter ``input_shape``.
+
+    filters: symbolic 4D tensor
+        Set of filters used in CNN layer of shape
+        (input channels, output channels, filter rows, filter columns).
+        See the optional parameter ``filter_shape``. **Note: the order for
+        ``output_channels`` and ``input_channels`` is reversed with respect to
+        ``conv2d``.**
+
+    output_shape: tuple/list of len 4 of int or Constant variable
+        The shape of the output of ``conv2d_transpose``.
+
+    filter_shape: None, tuple/list of len 4 of int or Constant variable
+        The shape of the filters parameter.
+        Optional, possibly used to choose an optimal implementation.
+        You can give ``None`` for any element of the list to specify that this
+        element is not known at compile time.
+
+    border_mode: str, int or tuple of two int
+        Refers to the ``border_mode`` argument of the corresponding forward
+        (non-transposed) convolution. See the argument description in
+        ``conv2d``.  What was ``padding`` for the forward convolution means
+        ``cropping`` the output of the transposed one. ``valid`` corresponds to
+        no cropping, ``full`` to maximal cropping.
+
+    input_dilation: tuple of len 2
+        Corresponds to ``subsample`` (also called strides elsewhere) in the
+        non-transposed convolution.
+
+    filter_flip: bool
+        If ``True``, will flip the filter rows and columns
+        before sliding them over the input. This operation is normally referred
+        to as a convolution, and this is the default. If ``False``, the filters
+        are not flipped and the operation is referred to as a cross-correlation.
+
+    filter_dilation: tuple of len 2
+        Factor by which to subsample (stride) the input.
+        Also called dilation elsewhere.
+
+    Returns
+    -------
+    Symbolic 4D tensor
+        Set of feature maps generated by the transposed convolution. Tensor is
+        of shape (batch size, output channels, output rows, output columns)
+
+    Notes
+    -----
+        If cuDNN is available, it will be used on the
+        GPU. Otherwise, it is the *CorrMM* convolution that will be used
+        "caffe style convolution".
+
+        This is only supported in Theano 0.8 or the development
```

Comment:
Also, we should probably at least mention the word "deconvolution" in here, so people looking for it will find the function.